.github/workflows/test.yml

```diff
@@ -0,0 +1,35 @@
+# GitHub Action to compile and test
+name: test
+on: [push, pull_request]
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        task: [data-check, codespell]
```

Comment:
Interesting use of matrix builds. I would've expected several steps/jobs instead of tasks defined by a matrix but why not?

Reply:
This was simply to get it the closest to our existing Travis setup, given it currently just calls the old script for that.

I think if you had steps/jobs, then where you've got say json-spec needed on more than one, you'd have to duplicate the install instructions (or install it even for the builds which don't need it, which doesn't seem very efficient to me.